lib/puppet/util/servicenow.rb

```diff
@@ -71,13 +71,18 @@ def settings(settings_file = Puppet[:confdir] + '/servicenow_reporting.yaml')
   end
   module_function :settings
 
-  def do_snow_request(uri, http_verb, body, user: nil, password: nil, oauth_token: nil)
+  def do_snow_request(uri, http_verb, body, user: nil, password: nil, oauth_token: nil, skip_cert_check: false)
     uri = URI.parse(uri)
 
+    opts = {
+      use_ssl: uri.scheme == 'https',
+    }
+
+    opts[:verify_mode] = OpenSSL::SSL::VERIFY_NONE if skip_cert_check
```

Comment:
I think `VERIFY_PEER` is the default for all supported ruby platforms, but you might want to always set it to be sure, and fall back to the insecure case if `skip_cert_check` is specified. Or use the puppet API below?